src/Akka.Streams.Kafka/Helpers/RestrictedConsumer.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Confluent.Kafka;
+
+namespace Akka.Streams.Kafka.Helpers
+{
+    /// <summary>
+    /// Offers parts of <see cref="IConsumer{K,V}"/> API which becomes available to <see cref="IPartitionEventHandler"/>
+    /// </summary>
+    public class RestrictedConsumer<K, V>
```

Comment:
This class is not used right now and exists to be used with `IPartitionEventHandler` not-yet-public interface to provide library users some safe ways to interact with kafka consumer.